sonarts-core/src/utils/navigation.ts

```diff
@@ -213,6 +213,16 @@ export function constructorOf(clazz: ts.ClassDeclaration | ts.ClassExpression):
   return clazz.members.find(member => member.kind === ts.SyntaxKind.Constructor) as ts.ConstructorDeclaration;
 }
 
+export function isArray(node: ts.Node, typeChecker: ts.TypeChecker): boolean {
+  const type = typeChecker.getTypeAtLocation(node);
+  return !!type.symbol && type.symbol.name === "Array";
+}
+
+export const ARRAY_MUTATING_CALLS = [
```

Comment:
@michalb-sonar I'm not super happy about having this set in this utils file, I would prefer small duplication as I doubt there will be more usages